code/modules/mob/living/simple_animal/bot/cleanbot.dm

```diff
@@ -80,9 +80,12 @@
 			user << "<span class='danger'>[src] buzzes and beeps.</span>"
 
 /mob/living/simple_animal/bot/cleanbot/process_scan(obj/effect/decal/cleanable/D)
+	. = list()
 	for(var/T in target_types)
 		if(istype(D, T))
-			return D
+			. += D
+	if(.)
+		. = pick(.)

```

Comment:
The entire logic of this PR actually seems wrong.
If D is a subtype of T it is placed on the list, yet there is only one D, so if it was going to be picked, it was going to be picked before, except if it is somehow MORE than 1 of those types it'll be in the list more times which only results in a higher chance to pick D instead of... picking D.

tl;dr wtf @Incoming5643